internal/extproc/translator/openai_awsbedrock.go

```diff
@@ -152,113 +180,194 @@ func (o *openAIToAWSBedrockTranslatorV1ChatCompletion) openAIToolsToBedrockToolC
+// the tool content is appended to the list of bedrock message list if tool_call is in openai message
+func (o *openAIToAWSBedrockTranslatorV1ChatCompletion) openAIMessageToBedrockMessageRoleUser(
+	openAiMessage *openai.ChatCompletionMessageParamUnion,
+) ([]*awsbedrock.Message, error) {
+	message := openAiMessage.Value.(openai.ChatCompletionUserMessageParam)
+	bedrockMessages := make([]*awsbedrock.Message, 0)
+
+	if _, ok := message.Content.Value.(string); ok {
+		bedrockMessages = append(bedrockMessages, &awsbedrock.Message{
+			Role: openAiMessage.Type,
+			Content: []*awsbedrock.ContentBlock{
+				{Text: ptr.To(message.Content.Value.(string))},
+			},
+		})
+	} else if contents, ok := message.Content.Value.([]openai.ChatCompletionContentPartUserUnionParam); ok {
+		chatMessage := &awsbedrock.Message{Role: openAiMessage.Type}
+		chatMessage.Content = make([]*awsbedrock.ContentBlock, 0, len(contents))
+		for _, contentPart := range contents {
+			if contentPart.TextContent != nil {
+				textContentPart := contentPart.TextContent
+				chatMessage.Content = append(chatMessage.Content, &awsbedrock.ContentBlock{
+					Text: &textContentPart.Text,
+				})
+			} else if contentPart.ImageContent != nil {
+				imageContentPart := contentPart.ImageContent
+				parts := strings.Split(imageContentPart.ImageURL.URL, ",")
+				if len(parts) == 2 {
+					formatPart := strings.Split(parts[0], ";")[0]
+					format := strings.TrimPrefix(formatPart, "data:image/")
+					chatMessage.Content = append(chatMessage.Content, &awsbedrock.ContentBlock{
+						Image: &awsbedrock.ImageBlock{
+							Format: format,
+							Source: awsbedrock.ImageSource{
+								Bytes: []byte(parts[1]),
+							},
+						},
+					})
+				} else {
+					return nil, fmt.Errorf("unexpected image data url")
+				}
+			}
+		}
+		bedrockMessages = append(bedrockMessages, chatMessage)
+	} else {
+		return nil, fmt.Errorf("unexpected content type")
+	}
+	// process tool_calls
+	for _, toolCall := range message.ToolCalls {
+		jsonBytes, err := json.Marshal(toolCall)
+		if err != nil {
+			return nil, err
+		}
+		bedrockMessages = append(bedrockMessages, &awsbedrock.Message{
+			Content: []*awsbedrock.ContentBlock{
+				{
+					ToolResult: &awsbedrock.ToolResultBlock{
+						ToolUseID: ptr.To(toolCall.ID),
+						Content: []*awsbedrock.ToolResultContentBlock{
+							{
+								JSON: ptr.To(string(jsonBytes)),
+							},
+						},
+					},
+				},
+			},
+		})
+	}
+	return bedrockMessages, nil
+}
+
+// openAIMessageToBedrockMessageRoleAssistant converts openai assistant role message
+// returns a list of bedrock message and error
+// the tool content is appended to the list of bedrock message list if tool_call is in openai message
+func (o *openAIToAWSBedrockTranslatorV1ChatCompletion) openAIMessageToBedrockMessageRoleAssistant(
+	openAiMessage *openai.ChatCompletionMessageParamUnion,
+) ([]*awsbedrock.Message, error) {
+	bedrockMessages := make([]*awsbedrock.Message, 0)
```

Comment:
same comment (ignore if not applicable) why is this pointer slice instead of value?)